### PR TITLE
80 Share Component

### PR DESCRIPTION
**Pull request by harleyjm**
#80 

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
Hey @harleyjm 
I'm unable to run the PR locally, it throws `Unable to resolve path to module 'react-share' ` so you need to `npm i react-share`
and also add `package.json` and `package-lock.json` to the PR

File: client/package.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
     "react-dom": "^17.0.1",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.1",
+    "react-share": "^4.3.1",
+    "storybook": "^6.1.15",
```
Comment (RamyAlshurafa) [inline client/package.json:18]:
```suggestion
```

why did you install this? i think its not needed and the required modules already listed in the devDependencies in the package.json

File: client/src/components/Share/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  TwitterIcon,
+  EmailShareButton,
+  FacebookShareButton,
+  TwitterShareButton,
+  WhatsappShareButton,
+} from 'react-share';
+
+import { Container, Icons, Text } from './style';
+
+const Share = ({ text, facebook, twitter, email, whatsapp }) => {
```
Comment (RamyAlshurafa) [inline client/src/components/Share/index.js:15]: you can give the text a defualt value
```suggestion
const Share = ({ text = 'Share this page', facebook, twitter, email, whatsapp }) => {
```

in this way, you don't have to give it that same value each time you use this component, and you can just overwrite it if you want a different text

File: client/src/components/Share/style.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import styled from '@emotion/styled';
+
+export const Container = styled('div')`
+  padding: 6px;
+  width: 100%;
+  height: 70px;
+  background: ${({ theme }) => `${theme.colors.gray} !important`};
+`;
+
+export const Text = styled('span')`
```
Comment (RamyAlshurafa) [inline client/src/components/Share/style.js:10]: you can use our common Typography component instead of this, this should be Body16 component

**Review by RamyAlshurafa — COMMENTED** [review]
Really nice work @harleyjm 
but left some comments for you
\+ the Twitter icon is not there in the design, and when you add all of them into component they disappear in the mobile because there is no enough space

image

File: client/package.json (hunk 1 above)
Comment (harleyjm) [reply client/package.json:18]: It's not needed, I think it was added when I was trying to correct a storybook error yesterday

File: client/src/components/Share/style.js (hunk 3 above)
Comment (harleyjm) [reply client/src/components/Share/style.js:10]: Sorry didn't realise there was a common typography component. I am still playing catchup having only looked at this yesterday.

**Review by RamyAlshurafa — APPROVED** [review]
Awsome @harleyjm , all perfect :)

**Pull request merged**